File: backend/app/models/verified_technician.py
from uuid import UUID
from typing import Dict, Any
from pydantic import BaseModel
from datetime import datetime


class BaseVerifiedTechnician(BaseModel):
    technician_id: UUID
    admin_id: UUID


class VerifiedTechnicianInDB(BaseVerifiedTechnician):
    created_at: datetime


class VerifiedTechnicianCreate(BaseVerifiedTechnician):
    class Config:
        json_schema_extra: Dict[str, Any] = {
            "example": {
                "technician_id": "123e4567-e89b-12d3-a456-426655440000",
                "admin_id": "123e4567-e89b-12d3-a456-426655440000",
            }
        }


class VerifiedTechnicianResponse(VerifiedTechnicianInDB):
    class Config:
        json_schema_extra: Dict[str, Any] = {
            "example": {
                "technician_id": "123e4567-e89b-12d3-a456-426655440000",
                "admin_id": "123e4567-e89b-12d3-a456-426655440000",
                "created_at": "2025-06-01T00:00:00",
            }
        }

File: frontend/src/lib/types/verified_technician.ts
import { z } from "zod";

const BaseVerifiedTechnicianSchema = z.object({
    client_id: z.string().uuid(),
    technician_id: z.string().uuid(),
});

export const VerifiedTechnicianCreateSchema = BaseVerifiedTechnicianSchema.extend({});

export const VerifiedTechnicianResponseSchema = BaseVerifiedTechnicianSchema.extend({
    created_at: z.string().datetime(),
});

export type VerifiedTechnicianCreate = z.infer<typeof VerifiedTechnicianCreateSchema>;
export type VerifiedTechnicianResponse = z.infer<typeof VerifiedTechnicianResponseSchema>;
